### PR TITLE
Create a way to localize strings

### PR DESCRIPTION
**Pull request by YodaForce157**
TODO

- [x] Create localized components
- [x] Implement those components
- [x] ~~Move over to Locale Enum?~~
- [x] ~~Find people to do translations/use google translate (its only words so it should be accurate until we get localizers)~~

### CODE REVIEW

File: src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java]: Rename to `I18n`

File: src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java]: Rename to `localize`

File: src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java]:
```suggestion
        LOGGER.debug("Getting localized string for key {}", key);
```

File: src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/LocalizationHandler.java]: Also, if its not found in english, this will be infinite recursion. So in that case you should just default to the `key` 
```suggestion
            return getLocalized(key);
```

File: src/main/resources/io/github/railroad/lang/en_us.properties
Comment (DaRealTurtyWurty) [inline src/main/resources/io/github/railroad/lang/en_us.properties]: Any reason for the spaces after `=`? Looks really weird

File: src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java]: This should use the enum.

File: src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java]: No need for this

File: src/main/java/io/github/railroad/utility/localization/Languages.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/Languages.java]: We could possibly consider just using `Locale` instead of this.

**Review by DaRealTurtyWurty — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/railroad/utility/localization/Languages.java
Comment (YodaForce157) [reply src/main/java/io/github/railroad/utility/localization/Languages.java]: I'm going to keep it using Languages for this PR, or atleast at the moment, I might move it over later but I'll prioritize updating all the components first.

**Comment by YodaForce157** [comment]
Ignore the conflict (I think) as I can't seem to do anything on my end.

**Comment by YodaForce157** [comment]
Should be OK now. I'll wait until this is merged/commented on to touch anything else because I'd rather not break everything _again_

Once merged I'll create another pr to finish up the few leftover components that haven't been localized and also finish up the language files.

File: src/main/java/io/github/railroad/Railroad.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/Railroad.java]:
```suggestion
```

File: src/main/java/io/github/railroad/Railroad.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/Railroad.java]: The title of the application should not be localized. 
```suggestion
        primaryStage.setTitle(localize("Railroad - 1.0.0(dev)");
```

File: src/main/java/io/github/railroad/project/ProjectSort.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/project/ProjectSort.java]: If this returns the key then the method should be called `getKey`. If it should still return the name then it should be localized.

File: src/main/java/io/github/railroad/project/ui/create/widget/ProjectSortComboBox.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/project/ui/create/widget/ProjectSortComboBox.java]: Any reason you can't use a `LocalizedListCell`?

File: src/main/java/io/github/railroad/settings/SettingsCategory.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/SettingsCategory.java]: If this returns the key then the method should be called `getKey`. If it should still return the name then it should be localized.

File: src/main/java/io/github/railroad/settings/ui/SettingsBehaviorPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsBehaviorPane.java]:
```suggestion
```

File: src/main/java/io/github/railroad/settings/ui/SettingsKeymapsPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsKeymapsPane.java]:
```suggestion
```

File: src/main/java/io/github/railroad/settings/ui/SettingsProjectsPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsProjectsPane.java]:
```suggestion
```

File: src/main/java/io/github/railroad/settings/ui/SettingsToolsPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsToolsPane.java]:
```suggestion
```

File: src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java]:
```suggestion
        var langBox = new RRVBox(10);
```

File: src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java]:
```suggestion
public class SettingsGeneralPane extends RRVBox {
```

File: src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/general/SettingsGeneralPane.java]:
```suggestion
import io.github.railroad.ui.defaults.RRVBox;
```

File: src/main/java/io/github/railroad/ui/localized/LocalizedComboBox.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/ui/localized/LocalizedComboBox.java]:
```suggestion
                if (object == null)
                    return "NullObject";

```

File: src/main/java/io/github/railroad/ui/localized/LocalizedComboBox.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/ui/localized/LocalizedComboBox.java]:
```suggestion
                if (key == null)
                    return "NullKey";

                return L18n.localize(key);
```

File: src/main/java/io/github/railroad/ui/localized/LocalizedComboBox.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/ui/localized/LocalizedComboBox.java]:
```suggestion
            }
            
            @Override
```

File: src/main/java/io/github/railroad/ui/localized/LocalizedLabel.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/ui/localized/LocalizedLabel.java]:
```suggestion
    private final AtomicReference<String> argString = new AtomicReference<>("");
```

File: src/main/java/io/github/railroad/ui/localized/LocalizedTooltip.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/ui/localized/LocalizedTooltip.java]:
```suggestion
    private final AtomicReference<String> argString = new AtomicReference<>("");
```

File: src/main/java/io/github/railroad/utility/localization/L18n.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/L18n.java]: Before we go too far with this, its probably worth having them be `.lang` extensions, not `.properties`

File: src/main/java/io/github/railroad/utility/localization/L18n.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/L18n.java]:
```suggestion
        }
        
        return LANG_CACHE.get(key).toString();
```

File: src/main/java/io/github/railroad/utility/localization/Languages.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/Languages.java]:
```suggestion
    EN_US("English"),
    ES_ES("Español"),
    FR_FR("Français"),
    DE_DE("Deutsch");
```

File: src/main/java/io/github/railroad/utility/localization/Languages.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/Languages.java]:
```suggestion
    public static Languages fromName(String name) {
        for(Languages lang : Languages.values()) {
            if(lang.name().equalsIgnoreCase(name)) {
                return lang;
            }
        }
        
        LOGGER.error("Cannot find language: {}", name);
        return null;
    }
```

File: src/main/java/io/github/railroad/utility/localization/Languages.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/Languages.java]: Rename to `Language`

File: src/main/java/io/github/railroad/welcome/WelcomeLeftPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/welcome/WelcomeLeftPane.java]: Window title should not be localized
```suggestion
        rightVbox.getChildren().add(new Label("Railroad IDE"));
```

File: src/main/java/io/github/railroad/welcome/WelcomeLeftPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/welcome/WelcomeLeftPane.java]: If this returns the key then the method should be called `getKey`. If it should still return the name then it should be localized.

File: themes/cupertino-dark.css
Comment (DaRealTurtyWurty) [inline themes/cupertino-dark.css]: You only need to do this in base.css. Not in all the themes.

**Review by DaRealTurtyWurty — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/railroad/project/ui/create/widget/ProjectSortComboBox.java
Comment (YodaForce157) [reply src/main/java/io/github/railroad/project/ui/create/widget/ProjectSortComboBox.java]: Because that isn't a dropdown? Unless I'm missing something. Changing it to a localized list cell just removes the dropdown part of it and messes with theming.

Comment (DaRealTurtyWurty) [reply src/main/java/io/github/railroad/project/ui/create/widget/ProjectSortComboBox.java]: We're doing `setButtonCell(new ListCell<>() {`

File: src/main/resources/io/github/railroad/styles/default-light.css
Comment (DaRealTurtyWurty) [inline src/main/resources/io/github/railroad/styles/default-light.css]:
```suggestion
```

File: src/main/java/io/github/railroad/utility/localization/L18n.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/L18n.java]:
```suggestion
        var newLang = Language.valueOf(ConfigHandler.getConfigJson().get("settings").getAsJsonObject().get("language").getAsString());
```

File: src/main/java/io/github/railroad/utility/localization/L18n.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/localization/L18n.java]:
```suggestion
        JsonObject updated = ConfigHandler.getConfigJson();
```

File: src/main/java/io/github/railroad/settings/ui/SettingsKeymapsPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsKeymapsPane.java]:
```suggestion
        var keyColumn = new TableColumn("Key");
        var actionColumn = new TableColumn("Action");
```

File: src/main/java/io/github/railroad/settings/ui/SettingsBehaviorPane.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsBehaviorPane.java]:
```suggestion
        var title = new LocalizedLabel("railroad.home.settings.behavior");
```

**Review by DaRealTurtyWurty — CHANGES_REQUESTED** [review]

**Review by DaRealTurtyWurty — APPROVED** [review]

File: src/main/java/io/github/railroad/project/ui/create/widget/ProjectSearchField.java
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/project/ui/create/widget/ProjectSearchField.java]: Since you're touching this, you might as well yeet this custom class for it. We don't need it to be it's own class

**Pull request merged**
